browser/src/view/components/Button/Button.js

```diff
@@ -1,40 +1,60 @@
 import React from 'react';
 import './Button.scss';
 
-export const Button = ({ label, style, icon }) => {
-  const renderButtonStyle = buttonStyle => {
-    switch (buttonStyle) {
-      case 'icon-only':
-        return icon;
-      case 'text-only':
-        return label;
-      case 'icon-text':
-        return (
-          <>
-            {icon} {label}
-          </>
-        );
-      default:
-        return null;
-    }
-  };
-  return <button className='btn-icon-container'>{renderButtonStyle(style)}</button>;
+export const Button = ({ layout, label, Icon, type }) => {
+  return (
+    <button
+      className={`${
+        type === ButtonAttributes.Type.destructive
+          ? 'btn-destructive'
+          : type === ButtonAttributes.Type.primary
+          ? 'btn-primary'
+          : 'btn-secondary'
+      }
+      ${
+        type === ButtonAttributes.Layout.iconAndLabel
+          ? 'btn-icon-label'
+          : type === ButtonAttributes.Layout.iconOnly
+          ? 'btn-icon-only'
+          : 'btn-label-only'
+      }`}
+    >
+      {layout === ButtonAttributes.Layout.iconOnly ||
+      layout === ButtonAttributes.Layout.iconAndLabel
+        ? Icon
+        : null}
+      {layout === ButtonAttributes.Layout.labelOnly ||
+      layout === ButtonAttributes.Layout.iconAndLabel
+        ? label
+        : null}
+    </button>
+  );
 };
 
-/*
-Button:
-- Style: icon only, text only, icon + text // type: normal, secondary, destructive
-- Interaction states: default, hover, selected (I don't think I'll worry about an :active/pressed button styling)
+const ButtonAttributes = {
+  Layout: {
+    iconOnly: 'icon',
+    labelOnly: 'label',
+    iconAndLabel: 'icon-label',
+  },
+  Type: {
+    primary: 'primary',
+    secondary: 'secondary',
+    destructive: 'destructive',
+  },
+};
+Object.freeze(ButtonAttributes);
 
-TODO:
-- Populate the navLinks list in App.js with an array of these button components instead
-- Translate the object in navLinks to instead have this component accept props:
-    + what Icon (svg) to display (how to make this optional if style === text only?)
-    + label (how to make this optional if style != text)
-    + style (icon/text/icon+text), and type (e.g. normal, secondary, alert, etc.)
-- Think of better names than 'style' and 'type'
-- Rename scss class names for this component to be more generic
-- Should the navLinks then live back in NavigationMenu? Or is that not good to have the object being rendered as well as the rendering logic in the same place?
-- In NavigationMenu.js, iterate over the array of components appropriately
-- Note to self: in the future, where will Route info live?
+Button.defaultProps = {
+  layout: ButtonAttributes.Layout.iconAndLabel,
+  label: undefined,
+  icon: undefined,
```

Comment:
Intentionally "undefined" types can be set to `null` to make it clear that the lack of value is intended.

Reply:
Thanks for clarifying that. I tried doing some research on undefined vs null but think I got it mixed up as to which one is best to use in this scenario.